src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -371,11 +395,28 @@ class DefaultColorControlSceneHandler : public scenes::DefaultSceneHandlerImpl
         }
     }
 
-    void AddAttributeValuePair(ScenesManagement::Structs::AttributeValuePair::Type * pairs, AttributeId id, uint32_t value,
+    /// AddAttributeValuePair
+    /// @brief Helper function to add an attribute value pair to the attribute value pair array in the color control SceneHandler
+    /// @param pairs list of attribute value pairs
+    /// @param id attribute id
+    /// @param value attribute value
+    /// @param attributeCount number of attributes in the list, incremented by this function, used to keep track of how many
+    /// attributes from the array are being used for the list to encode
+    template <typename Type>
+    void AddAttributeValuePair(ScenesManagement::Structs::AttributeValuePairStruct::Type * pairs, AttributeId id, Type value,
                                size_t & attributeCount)
     {
-        pairs[attributeCount].attributeID    = id;
-        pairs[attributeCount].attributeValue = value;
+        static_assert(sizeof(Type) == sizeof(uint8_t) || sizeof(Type) == sizeof(uint16_t), "Type must be uint8_t or uint16_t");
```

Comment:
This should be an `is_same_v` assert, no?  Again, otherwise if something signed gets added this function will silently do the wrong thing.....

Ideally, you would just static_assert in the "else" of the if constexpr cascade, but C++ does not seem to allow that.  :(